libraries/src/AWS.Lambda.Powertools.Metrics/Model/MetricDirective.cs

```diff
@@ -141,7 +141,13 @@ public void AddMetric(string name, double value, MetricUnit unit, MetricResoluti
         {
             var metric = Metrics.FirstOrDefault(m => m.Name == name);
             if (metric != null)
-                metric.AddValue(value);
+            {
+                if(metric.Values.Count < PowertoolsConfigurations.MaxMetrics)
+                    metric.AddValue(value);
+                else
```

Comment:
I don't believe we will ever be in this situation (it only accepts one double not an array) but added the throwing exception for consistency with the bellow throw